File: app/src/main/java/net/tlalka/android/fiszki/view/elements/PageElement.java
package net.tlalka.android.fiszki.view.elements;

import net.tlalka.android.fiszki.R;
import net.tlalka.android.fiszki.view.activities.BasePageActivity;
import net.tlalka.android.fiszki.view.activities.HelpActivity;
import net.tlalka.android.fiszki.view.activities.LessonListActivity;
import net.tlalka.android.fiszki.view.activities.TestListActivity;
import net.tlalka.android.fiszki.view.activities.WordsActivity;

public enum PageElement {

    LESSONS(R.string.nav_lessons, LessonListActivity.class),
    TESTS(R.string.nav_tests, TestListActivity.class),
    WORDS(R.string.nav_words, WordsActivity.class),
    HELP(R.string.nav_help, HelpActivity.class);

    private final int resourceId;
    private final Class<? extends BasePageActivity> activityClass;

    PageElement(int resourceId, Class<? extends BasePageActivity> activityClass) {
        this.resourceId = resourceId;
        this.activityClass = activityClass;
    }

    public Integer getResourceId() {
        return this.resourceId;
    }

    public Class<?> getActivityClass() {
        return this.activityClass;
    }
}

File: app/src/main/java/net/tlalka/android/fiszki/view/elements/PageElement.kt
package net.tlalka.android.fiszki.view.elements

import android.app.Activity
import net.tlalka.android.fiszki.R
import net.tlalka.android.fiszki.view.activities.HelpActivity
import net.tlalka.android.fiszki.view.activities.LessonListActivity
import net.tlalka.android.fiszki.view.activities.TestListActivity
import net.tlalka.android.fiszki.view.activities.WordsActivity
import kotlin.reflect.KClass

enum class PageElement(val resourceId: Int, private val activityClass: KClass<out Activity>) {

    LESSONS(R.string.nav_lessons, LessonListActivity::class),
    TESTS(R.string.nav_tests, TestListActivity::class),
    WORDS(R.string.nav_words, WordsActivity::class),
    HELP(R.string.nav_help, HelpActivity::class);

    fun getActivityClass(): Class<out Activity> {
        return activityClass.java
    }
}
